### PR TITLE
Enable passing in model instance to @can directive

### PR DESCRIPTION
**Pull request by eriktisme**
- [x] Added or updated tests
- [x] Added Docs for all relevant versions

**Related Issue/Intent**
Enable passing model instance to determine if a user is authorized to perform a given action on this model.

Resolves https://github.com/nuwave/lighthouse/issues/244

**Changes**
Check if an ID is passed in the query when `@can` directive is given.

**Breaking changes**
No breaking changes to my knowledge.

### CODE REVIEW

File: src/Schema/Directives/BaseDirective.php [hunk 1]
```diff
@@ -125,6 +125,25 @@ protected function getModelClass(string $argumentName = 'model'): string
...
     }
 
+    /**
+     * Find the model instance by the given key.
+     *
+     * @param string|int|null $id
+     * @return \Illuminate\Database\Eloquent\Model|null
+     * @throws \Nuwave\Lighthouse\Exceptions\DirectiveException
+     */
+    protected function findModelById($id = null): ?Model
```
Comment (spawnia) [inline src/Schema/Directives/BaseDirective.php]: Why not just use `$modelClass::find()` here?

Comment (eriktisme) [reply src/Schema/Directives/BaseDirective.php]: You are right! I first passed in the whole array of arguments. Now I'm directly passing in the ID `find` is indeed better

**Comment by spawnia** [comment]
Hm, what about `@can` in scenarios with multiple models? How does Laravel handle that?

**Comment by eriktisme** [comment]
> Hm, what about `@can` in scenarios with multiple models? How does Laravel handle that?

I myself have never used one with multiple models in one check. I will check, one moment!

**Comment by spawnia** [comment]
@eriktisme seems like it is not supported by Laravel anyway. I think this PR should be good to go.

@chrissm79 what do you think?

**Comment by eriktisme** [comment]
@spawnia Thank you, you too!

**Comment by chrissm79** [comment]
Great job @eriktisme 👍Thanks for this!

**Comment by acro5piano** [comment]
Thank you @eriktisme !

**Comment by BasMulders** [comment]
This is great. Thanks a lot! 🙌

**Comment by plunkettscott** [comment]
👏 A much-needed addition. Thanks @eriktisme!

**Pull request merged**